### ISSUE TITLE
Make Sensors 6211/1.1 , 6211/1.2, 6211/2.1, 6211/2.2, 6251/0.4, 6221/2.0 available as Switches

### ISSUE THREAD

**Issue by floriangrotz** [pgerke/homebridge-freeathome-local-api#41]
Make Sensors available as Switches - so that I can use a Sensor switch as a Switch in Homekit to be able to control other Homekit Devices

**Comment by pgerke** [comment]
To make a new accessory available in the plugin I'd need the function ID and the list of input and output datapoints with their pairing IDs.

**Comment by korneliuscode** [comment]
@pgerke can you provide a quick guide on where/how to get that info?

**Comment by pgerke** [comment]
@korneliuscode I have started to create a guide [here](https://github.com/pgerke/homebridge-freeathome-local-api/wiki/Retrieving-the-Configuration-for-a-device-from-the-System-Access-Point), hope it helps!

**Comment by korneliuscode** [comment]
> @korneliuscode I have started to create a guide [here](https://github.com/pgerke/homebridge-freeathome-local-api/wiki/Retrieving-the-Configuration-for-a-device-from-the-System-Access-Point), hope it helps!

This is great. Thanks a lot. I just posted one on the other issue.

**Comment by floriangrotz** [comment]
Also here the 
6211/1.1
[Switch_1_1_response_1672577386034.json.txt](https://github.com/pgerke/homebridge-freeathome-local-api/files/10328791/Switch_1_1_response_1672577386034.json.txt)

6211/2.2
[Switch_2_2_response_1672577458845.json.txt](https://github.com/pgerke/homebridge-freeathome-local-api/files/10328793/Switch_2_2_response_1672577458845.json.txt)

**Comment by pgerke** [comment]
@korneliuscode Moved here from #30:

@pgerke thanks for the guide! Here are the 2 snippets
`6251/0.4`:
```
{
  "00000000-0000-0000-0000-000000000000": {
    "devices": {
      "XXXXXXX": {
        "interface": "TP",
        "displayName": "Switch actuator, 4gang, 16 A, MDRC",
        "unresponsive": false,
        "channels": {
          "ch0000": {
            "floor": "01",
            "room": "01",
            "displayName": "Tor 1",
            "functionID": "45",
            "inputs": {
              "idp0001": {
                "pairingID": 2,
                "value": "1"
              },
              "idp0003": {
                "pairingID": 4,
                "value": "0"
              }
            },
            "outputs": {}
          },
          "ch0001": {
            "floor": "01",
            "room": "01",
            "displayName": "Tor 2",
            "functionID": "45",
            "inputs": {
              "idp0001": {
                "pairingID": 2,
                "value": "1"
              },
              "idp0003": {
                "pairingID": 4,
                "value": "0"
              }
            },
            "outputs": {}
          },
          "ch0002": {
            "displayName": "Ⓒ",
            "functionID": "7",
            "inputs": {
              "idp0000": {
                "pairingID": 1,
                "value": "0"
              },
              "idp0001": {
                "pairingID": 2,
                "value": "0"
              },
              "idp0002": {
                "pairingID": 3,
                "value": "0"
              },
              "idp0003": {
                "pairingID": 4,
                "value": "0"
              },
              "idp0004": {
                "pairingID": 6,
                "value": "0"
              }
            },
            "outputs": {
              "odp0000": {
                "pairingID": 256,
                "value": "0"
              },
              "odp0001": {
                "pairingID": 257,
                "value": "0"
              },
              "odp0004": {
                "pairingID": 273,
                "value": "0"
              }
            }
          },
          "ch0003": {
            "displayName": "Ⓓ",
            "functionID": "7",
            "inputs": {
              "idp0000": {
                "pairingID": 1,
                "value": "0"
              },
              "idp0001": {
                "pairingID": 2,
                "value": "0"
              },
              "idp0002": {
                "pairingID": 3,
                "value": "0"
              },
              "idp0003": {
                "pairingID": 4,
                "value": "0"
              },
              "idp0004": {
                "pairingID": 6,
                "value": "0"
              }
            },
            "outputs": {
              "odp0000": {
                "pairingID": 256,
                "value": "0"
              },
              "odp0001": {
                "pairingID": 257,
                "value": "0"
              },
              "odp0004": {
                "pairingID": 273,
                "value": "0"
              }
            }
          }
        }
      }
    }
  }
}
```

And the wall mounted sensor unit `6221/2.0` which actually triggers it

```
{
  "00000000-0000-0000-0000-000000000000": {
    "devices": {
      "YYYYYYYYYYY": {
        "floor": "01",
        "room": "01",
        "interface": "TP",
        "displayName": "Schalter Zentral ",
        "unresponsive": false,
        "channels": {
          "ch0001": {
            "displayName": "Schalter Zentral ",
            "functionID": "6",
            "inputs": {
              "idp0005": {
                "pairingID": 273,
                "value": "0"
              },
              "idp000b": {
                "pairingID": 278,
                "value": "0"
              },
              "idp000c": {
                "pairingID": 18,
                "value": "1"
              }
            },
            "outputs": {
              "odp0007": {
                "pairingID": 4,
                "value": "0"
              }
            }
          },
          "ch0002": {
            "displayName": "Schalter Zentral ",
            "functionID": "6",
            "inputs": {},
            "outputs": {
              "odp0007": {
                "pairingID": 4,
                "value": "0"
              }
            }
          },
          "ch0004": {
            "displayName": "Schalter Zentral ",
            "functionID": "4",
            "inputs": {
              "idp0000": {
                "pairingID": 256,
                "value": "0"
              },
              "idp0005": {
                "pairingID": 273,
                "value": "0"
              },
              "idp0006": {
                "pairingID": 261,
                "value": "0"
              },
              "idp000b": {
                "pairingID": 278,
                "value": "0"
              },
              "idp000c": {
                "pairingID": 18,
                "value": "1"
              }
            },
            "outputs": {
              "odp0005": {
                "pairingID": 2,
                "value": "1"
              },
              "odp0007": {
                "pairingID": 4,
                "value": "0"
              }
            }
          },
          "ch0005": {
            "displayName": "Schalter Zentral ",
            "functionID": "4",
            "inputs": {
              "idp0000": {
                "pairingID": 256,
                "value": "0"
              },
              "idp0006": {
                "pairingID": 261,
                "value": "0"
              }
            },
            "outputs": {
              "odp0005": {
                "pairingID": 2,
                "value": "1"
              },
              "odp0007": {
                "pairingID": 4,
                "value": "0"
              }
            }
          }
        }
      }
    }
  }
}
```

**Comment by korneliuscode** [comment]
@pgerke Do you want to build a pre package including the recent changes for testing?

**Comment by pgerke** [comment]
No problem, here you go [v1.5.0-pre.4](https://www.npmjs.com/package/homebridge-freeathome-local-api/v/1.5.0-pre.4) 😉

**Comment by korneliuscode** [comment]
Great thank you. The bridge shows the right number of devices. But the list only shows 51 in HomeKit (instead of 68). Is there any Limitation in HomeKit for device types?
![1679B71F-2E2C-4F49-BED2-E5980E8C3108](https://user-images.githubusercontent.com/7031553/211135180-66e89070-34d6-4018-8c08-88176b4c058f.jpeg)

**Comment by pgerke** [comment]
If there is a limitation, it is not the issue here. I have over a hundred devices connected to my productive Homebridge instance and it's working without issue. I'd start troubleshooting with a restart of the Home App. I've also made a habit of closing the Home App while I install updates, because I have the impression that having the app open while devices are being recognized causes issues, but I don't have any reliable data on that one. Good luck! 🤞

**Comment by korneliuscode** [comment]
I figured it out. The devices are added to the Standardraum, but you have to go in that room to see them at the bottom. Turns out the new equipment needs a home central. Do you know anything about that?
![Uploading E054BA60-253D-4509-AE18-459984D3285F.png…]()

**Comment by korneliuscode** [comment]
I figured it out. The devices are added to the Standardraum, but you have to go in that room to see them at the bottom. Turns out the new equipment needs a home central. Do you know anything about that?
![478DAF45-2166-4F2A-A1C0-07A40353C4BD](https://user-images.githubusercontent.com/7031553/211166535-8fecf01e-cb07-438f-b955-32fbc3074616.png)

**Comment by pgerke** [comment]
I was not aware of the restriction, but it kinda makes sense. The sensors are only useful, if you use them as a trigger for actions. Without a permanently available control center, there might not be a device present at home that can trigger the action. Additional information is available with Apple: https://support.apple.com/de-de/HT207057

**Comment by pgerke** [comment]
Yesterday I created the new accessories for the FID_STAIRCASE_LIGHT_SENSOR (FID 4), the FID_SCENE_SENSOR (FID 6) and the FID_TRIGGER (FID 45). Devices are detected correctly in Apple Home, but the SysAP is not sending a signal when I trigger the sensor. I can see in my monitor and in the System Access Point analysis tool that the scenes and the involved devices are activated as expected, but no message is sent for the scene controller. The same is true for the staircase light sensor. For the FID_TRIGGER I can't say as I don't have a device with that FID in my setup. 

I'd rather not change the scope of the ticket, so I'll probably add the accessories as experimentally supported. Maybe B+J changes the behavior in a firmware update in the future.

**Comment by pgerke** [comment]
Support for scene sensors is included in preview version [v1.5.0-pre.6](https://www.npmjs.com/package/homebridge-freeathome-local-api/v/1.5.0-pre.6).
